tests/linalg_tests.cpp

```diff
@@ -0,0 +1,306 @@
+// Copyright © 2023 Apple Inc.
+
+#include "doctest/doctest.h"
+
+#include <cmath>
+#include <iostream>
+#include "mlx/linalg.h"
+#include "mlx/mlx.h"
+
+using namespace mlx::core;
+using namespace mlx::core::linalg;
+
+TEST_CASE("[mlx.core.linalg.norm] no ord") {
```

Comment:
this file eventually is going to be long when adding other linalg tests. i have an upcoming PR that will potentially add various cases to test QR factorization.

should lingalg test files be prefixed along with it's operation name? for ex: `linalg_norm_tests.cpp`
`linalg_qr_tests.cpp`
`linalg_svd_tests.cpp`

Reply:
I agree. I think we should separate the tests when necessary. The similar holds for `test_nn` for example.